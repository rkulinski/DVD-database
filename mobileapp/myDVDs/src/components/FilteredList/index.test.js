import React from 'react';
import { shallow } from 'enzyme';
import { FilteredList } from './';


describe('Testing FilteredList component', () => {
  describe('Default props', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<FilteredList/>);
    });
    it('renders as expected', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('With provided list', () => {
    const renderItemFn = jest.fn();
    const props = {
      itemsArray: [
        'abc',
        'def',
        'ghi',
      ],
      renderItem: renderItemFn,
    };

    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<FilteredList {...props}/>);
    });

    it('renders FlatList', () => {
      expect(wrapper.find('FlatList').length).toEqual(1);
    });
    it('renders all items', () => {
      expect(wrapper.find('FlatList').at(0).prop('data').length).toEqual(props.itemsArray.length);
    });
    it('filters based on input', () => {
      wrapper.instance().updateList('ab');
      expect(wrapper.state('searchTerm')).toEqual('ab');
      expect(wrapper.state('filteredList')).toEqual(['abc']);
    });
    it('should have empty list when nothing mathing', () => {
      wrapper.instance().updateList('qweqwe');
      expect(wrapper.state('filteredList')).toEqual([]);
    });
  });
});
