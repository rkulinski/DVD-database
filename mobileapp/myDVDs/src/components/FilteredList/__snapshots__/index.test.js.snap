// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing FilteredList component Default props renders as expected 1`] = `
<Component
  style={
    Array [
      Object {
        "width": "100%",
      },
    ]
  }
>
  <Component
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Search..."
      style={
        Array [
          Object {
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "fontSize": 37.5,
            "height": 40,
            "marginLeft": 5,
            "padding": 10,
            "width": "80%",
          },
        ]
      }
      value=""
    />
    <TouchableOpacity
      activeOpacity={0.2}
      style={
        Array [
          Object {
            "justifyContent": "center",
            "width": "20%",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 30,
              "textAlign": "center",
            },
          ]
        }
      >
        +
      </Text>
    </TouchableOpacity>
  </Component>
  <FlatList
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</Component>
`;
